### PR TITLE
qt5: Fix build with clang/musl

### PR DESCRIPTION
**Pull request by kraj**
Signed-off-by: Khem Raj <raj.khem@gmail.com>

### CODE REVIEW

**Comment by otavio** [comment]
I think it is good to go. Let's wait for the other review ...

File: recipes-qt/qt5/qtbase_git.bb [hunk 1]
```diff
@@ -33,6 +33,7 @@ SRC_URI += "\
     file://0015-double-conversion-support-AARCH64EB-and-arm-BE.patch \
     file://0016-Disable-ltcg-for-host_build.patch \
     file://0017-Qt5GuiConfigExtras.cmake.in-cope-with-variable-path-.patch \
+    file://0020-corelib-Include-sys-types.h-for-uint32_t.patch \
```
Comment (shr-project) [inline recipes-qt/qt5/qtbase_git.bb]: What about nativesdk-qtbase? I'll add it there when cherry-picking to master-next and thud-next.

File: recipes-qt/maliit/maliit-framework-qt5_git.bb [hunk 2]
```diff
@@ -53,6 +53,7 @@ EXTRA_QMAKEVARS_PRE = "\
     CONFIG+=local-install \
     CONFIG+=nosdk \
     CONFIG+=nodoc \
+    CONFIG+=notests \
```
Comment (shr-project) [inline recipes-qt/maliit/maliit-framework-qt5_git.bb]: I haven't seen them failing with gcc, can we disable them only for clang? Or have you seen failing them like this with gcc?

**Comment by shr-project** [comment]
Cherry-picked with suggested modifications to master-next and thud-next for build tests.

File: recipes-qt/maliit/maliit-framework-qt5_git.bb (hunk 2 above)
Comment (otavio) [reply recipes-qt/maliit/maliit-framework-qt5_git.bb]: It is indeed a valid point, is it possible to "fix" the tests?

Comment (kraj) [reply recipes-qt/maliit/maliit-framework-qt5_git.bb]: lets change it to 

```
EXTRA_QMAKEVARS_PRE_append_toolchain-clang = " notests"
```
for now.

File: recipes-qt/qt5/qtbase_git.bb (hunk 1 above)
Comment (kraj) [reply recipes-qt/qt5/qtbase_git.bb]: yes we can add it to native as well. this change is general across qtbase

**Comment by shr-project** [comment]
Pushed to master and thud, thanks!
   9fc65c0..201fcf2  master -> master
   010a29e..201fcf2  thud -> thud

**Pull request closed without merging**